modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/ranges.scala

```diff
@@ -0,0 +1,86 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.data.{NonEmptyList, ValidatedNel}
+import cats.syntax.all._
+
+import scala.util.control.NoStackTrace
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.cats._
+import eu.timepit.refined.types.numeric.NonNegLong
+
+@derive(eqv, show)
+sealed trait RangesError extends NoStackTrace
+
+case class InvalidInput(str: String) extends RangesError {
+  override def getMessage = s"Invalid input: $str"
+}
+
+case class InvalidRange(error: RangeError) extends RangesError {
+  override def getMessage: String = error.getMessage
+}
+
+case class ReversedRange(start: NonNegLong, end: NonNegLong) extends RangesError {
+  override def getMessage = s"The start, $start, cannot be greater than the end, $end"
+}
+
+object ranges {
+
+  type RangesValidationResult[A] = ValidatedNel[RangesError, A]
+
+  private val delimiter = ";"
+
+  def decode(input: String): RangesValidationResult[NonEmptyList[Range]] =
+    Either
+      .cond(
+        input.nonEmpty,
+        input,
+        InvalidInput(input)
+      )
+      .toValidatedNel
```

Comment:
I noticed this pattern repeated a few times. FYI, you can make the Validated directly with `Validated.condNel`

Reply:
Replaced `Either.cond(...).toValidatedNel` with `Validated.condNel`.